### DISCUSSION TITLE
Possible to split tunnel based on IP with Wireguard?

### DISCUSSION THREAD

**Discussion by PhysicsIsAwesome** [celzero/rethink-app#1724] [Q&A] [answered]
Hi,

first of all, thank you for this awesome app. I would like to know if it is possible to split tunnel based on IP with Wireguard in the RethinkDNS app?

Use case is as follows. I would like to route all traffic (including all DNS-traffic, since my DNS server is within this range) within a certain private IP range via Wireguard to my router at home, primarily to access my home server. I would also like to use a commercial VPN to hide my IP for all IPs outside of above mentioned  ip range. This should happen for the same apps. Theoretically I could achieve this by routing everything first to my home network, then everything not meant for my home network to the commercial VPN via policy-based routing configured in my home router. Problem is, this would give high latency, slow speed and is cumbersome. It would be better to directly split tunnel within the RethinkDNS app, depending on IP range, for example with two Wireguard configuration files, each with their own interface and peer, but with non-overlapping "Allowed IPs".

**Comment by ignoramous** [comment, marked as answer]
Not possible today. We could add it but it'll cause more confusion for other users.

- #1697
